api/src/main/java/io/opentelemetry/common/ReadableKeyValuePairs.java

```diff
@@ -16,39 +16,28 @@
 
 package io.opentelemetry.common;
 
-import javax.annotation.Nullable;
-
 /**
  * A read-only container for String-keyed attributes.
  *
  * @param <V> The type of the values contained in this.
  */
-public interface ReadableKeyValuePairs<V> {
+public interface ReadableKeyValuePairs<K, V> {
```

Comment:
I can't find this type parameter used in #1598. Why do we need a type parameter for the key? Don't we always use `String` for the key? Accessors may use specialized types for type safety though

```
Attributes {
  void set(StringKey, String value)
  void set(BooleanKey, boolean value)
  void set(Key<T>, T value)

  String getString(StringKey);
  Boolean getBoolean(BooleanKey);
}
```

I don't see a generic parameter for the key needed here, am I missing anything?

Also, I realized the typed keys isn't great for getters 😅 It might be a reason why in grpc-context, the "setters" (e.g., Context.withValue) accept key as a parameter, and getting is something like `MyKey.get(context)`. Since we don't have getters though not a big deal just noticed it.

Reply:
This will be needed when we switch the key type, since they won't be strings any more. The prototype didn't mess with the underlying implementation, but just stuck with Maps for demo simplicity (note the TODOs in the prototype code).

Reply:
I updated the PR description with some details that I realized was missing for people other than me. :)